### PR TITLE
[jenkins] feat: add jenkins support

### PR DESCRIPTION
**Pull request by Wayonb**
Add Jenkins pipeline support.  This reuse the shared library pipeline.

### CODE REVIEW

**Comment by segfaultxavi** [comment]
I am afraid I have no experience with Jenkins, I won't be able to provide a meaningful review.
If the CI works and it integrates nicely with GitHub... I won't complain 😊

File: .gitlintmodules [hunk 1]
```diff
@@ -0,0 +1 @@
+symbol/docs|jenkins
```
Comment (Jaguar0625) [inline .gitlintmodules:1]: probably just 'docs' is fine

File: init.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+
```
Comment (Jaguar0625) [inline init.sh:1]: can delete first blank line

File: scripts/ci/setup_build.sh [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+set -ex
+
+pip install -r requirements.txt
```
Comment (Jaguar0625) [inline scripts/ci/setup_build.sh:5]: if there is py code should you be running py linter too ?

File: scripts/ci/transifex.sh [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+tx_push() {
+    tx push --source --parallel
+}
+
+echo "Publishing to Transifex..."
+tx_install
+tx_init
+tx_push
```
Comment (Jaguar0625) [inline scripts/ci/transifex.sh:24]: not sure if you need functions vs just inlining

File: scripts/ci/setup_build.sh (hunk 3 above)
Comment (segfaultxavi) [reply scripts/ci/setup_build.sh:5]: The only Python code is Sphinx and its extensions (the docs engine).

File: scripts/ci/transifex.sh (hunk 4 above)
Comment (Wayonb) [reply scripts/ci/transifex.sh:24]: yea, meant to move these to inline

File: scripts/ci/setup_build.sh (hunk 3 above)
Comment (Wayonb) [reply scripts/ci/setup_build.sh:5]: Are we modifying these files directly?

https://github.com/symbol/symbol-docs/blob/3c8e3acf21c67eec066e3c35f36a02e6e4f33a72/source/_ext/card.py#L3

Comment (segfaultxavi) [reply scripts/ci/setup_build.sh:5]: Yeah, that's a Sphinx extension I wrote. Feel free to lint anything in the `_ext` folder but, in theory, those files should rarely be modified.

Comment (Wayonb) [reply scripts/ci/setup_build.sh:5]: Started fixing python linting errors.  The more I fix the more shows up :(

**Review by Jaguar0625 — APPROVED** [review]

**Pull request merged**